app/code/community/Oyst/OneClick/Model/Magento/Quote.php

```diff
@@ -241,14 +241,23 @@ private function getAddressData()
         $street = isset($customerAddress['street']) ? $customerAddress['street'] : '';
         $street .= isset($customerAddress['complementary']) ? ' ' . $customerAddress['complementary'] : '';
 
+        $city = null;
+        if (isset($customerAddress['postcode'])) {
+            $city = $this->mapCityZipCode($customerAddress['postcode']);
+
+            if (is_null($city)) {
```

Comment:
this if block have to be outside of `if (isset($customerAddress['postcode'])) {`